src/selectors/entities/posts.js

```diff
@@ -251,29 +251,19 @@ export function makeGetPostsAroundPost() {
                 return null;
             }
 
-            const focusedPostIndex = postIds.indexOf(focusedPostId);
-            if (focusedPostIndex === -1) {
-                return null;
-            }
-
-            const desiredPostIndexBefore = focusedPostIndex - (Posts.POST_CHUNK_SIZE / 2);
-            const minPostIndex = desiredPostIndexBefore < 0 ? 0 : desiredPostIndexBefore;
-
-            const slicedPostIds = postIds.slice(minPostIndex);
-
             const posts = [];
             const joinLeavePref = myPreferences[getPreferenceKey(Preferences.CATEGORY_ADVANCED_SETTINGS, 'join_leave')];
             const showJoinLeave = joinLeavePref ? joinLeavePref.value !== 'false' : true;
 
-            for (let i = 0; i < slicedPostIds.length; i++) {
-                const post = allPosts[slicedPostIds[i]];
+            for (let i = 0; i < postIds.length; i++) {
```

Comment:
I'm a little worried about the performance of this for a channel that's accumulated a large number of posts. What about doing something similar to https://github.com/saturninoabril/mattermost-redux/blob/2f4df765d377f1052d7f2dc90f45c7aa7412fc87/src/selectors/entities/posts.js#L225 where we pass in the number of posts to display. In this case I think you'd need the number of posts before and after to display

That way we're not formatting 5000 posts when the user can only see 100 or so

Reply:
actually @jwilander brings a very good point, and I totally missed it